zerver/forms.py

```diff
@@ -50,13 +50,15 @@ def email_is_not_mit_mailing_list(email):
                 raise
 
 class RegistrationForm(forms.Form):
+    MAX_PASSWORD_LENGTH = 100
+    MAX_REALM_SUBDOMAIN_LENGTH = 40
```

Comment:
This you should be in Realm as well -- it's the length of this field:

    string_id = models.CharField(max_length=40, unique=True) # type: Text